framework/spec.go

```diff
@@ -29,81 +81,52 @@ type Resource interface {
-	GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error)
-	// Name returns the resource's name used for identification.
-	Name() string
-	// ProcessCreateState receives the new custom object observed during TPR
-	// watches. It also receives the state intended to be created as provided by
-	// GetCreateState. ProcessCreateState only has to create resources based on
-	// its provided input. All other reconciliation logic and state transformation
-	// is already done at this point of the reconciliation loop.
-	//
-	// NOTE ProcessCreateState is called on create and update events. When called
-	// on create events the provided custom object will be the custom object
-	// currently known to the informer. On update events the informer knows about
-	// the old and the new custom object. ProcessCreateState then receives the new
-	// custom object to be able to process the create state of a system.
-	ProcessCreateState(ctx context.Context, obj, createState interface{}) error
-	// ProcessDeleteState receives the new custom object observed during TPR
-	// watches. It also receives the state intended to be deleted as provided by
-	// GetDeleteState. ProcessDeleteState only has to delete resources based on
-	// its provided input. All other reconciliation logic and state transformation
-	// is already done at this point of the reconciliation loop.
-	//
-	// NOTE ProcessDeleteState is called on delete and update events. When called
-	// on delete events the provided custom object will be the custom object
-	// currently known to the informer. On update events the informer knows about
-	// the old and the new custom object. ProcessDeleteState then receives the new
-	// custom object to be able to process the delete state of a system.
-	ProcessDeleteState(ctx context.Context, obj, deleteState interface{}) error
-	// ProcessUpdateState receives the new custom object observed during TPR
-	// watches. It also receives the state intended to be updated as provided by
-	// GetUpdateState. ProcessUpdateState has to update resources based on its
-	// provided input. All other reconciliation logic and state transformation is
-	// already done at this point of the reconciliation loop.
-	ProcessUpdateState(ctx context.Context, obj, updateState interface{}) error
-	// Underlying returns the underlying resource which is wrapped by the calling
-	// resource. Underlying must always return a non nil resource. Otherwise
-	// proper resource chaining and execution cannot be guaranteed. In case a
-	// resource does not wrap any other resource, Underlying must return the
-	// resource that does not wrap any resource. The returned resource is then the
-	// origin, the underlying resource of the chain. In combination with Name,
-	// Underlying can be used for proper identification.
-	Underlying() Resource
+
+	// NewUpdatePatch is callend upon observed CRO/TPO change. It receives
```

Comment:
I'm not sure if referencing CRO and TPO (or TPR later) is ok,TPOs are already deprecated and CRO may change in the future, IMO custom resource/object is more generic and difficult to get stale, WDYT?